WizardEngine/Entity.h

```diff
@@ -77,5 +49,16 @@ class Entity : public Object
 	virtual bool Update(float deltaTime);
 
 	static void HandleCollision(Collider*, Collider*);
+
+	Mesh* mesh = nullptr;
```

Comment:
I think mesh, material and velocity should just be public instead of having accessors which are bulky. The entity shouldn't care if the velocity or material changes.